bot/common/threads/shared_steps.py

```diff
@@ -36,3 +36,15 @@ async def handle_emoji(self, raw_reaction):
         if not selected_guild_reaction:
             raise Exception("Reacted with the wrong emoji")
         return None, None
+
+
+class EmptyStep(BaseStep):
```

Comment:
Interesting the builder already sort of has a concept for this which is probably poorly documented, but it is the END step. A dev can implement a control hook step which runs after the send and handle emoji functions and ends the thread. An example can be found here.

https://github.com/Govrn-HQ/KevinMalone/blob/0c35a2ad9555e7e4f6f2810b4f2a148f921f90ae/bot/common/threads/onboarding.py#L207